### PR TITLE
Miner, slashing, reputation, and ask services

### LINKED ISSUE #11: gRPC service impl for miners, slashing, and reputation
Exposing these services will give us the flexibility to use the data in many places including the fil-tools cli, textile, and web services served by fil-tools and/or textile.

### PR DESCRIPTION
**Pull request by asutula**
Just building out the gRPC services for all the indexes and the reputation module. Client code coming in the next chunk of work.

Closes #11

### CODE REVIEW

File: api/server/server.go [hunk 1]
```diff
@@ -36,18 +41,27 @@ var (
...
+	dealsService      *deals.Service
+	walletService     *wallet.Service
+	reputationService *reputation.Service
+	askService        *ask.Service
+	minerService      *miner.Service
+	slashingService   *slashing.Service
+
+	rpc *grpc.Server
+
+	closeLotus func()
```
Comment (asutula) [inline api/server/server.go]: Just did some organizing here and below to keep things visually clear and consistent.

File: index/ask/pb/ask.proto [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+message QueryReply {
+    repeated StorageAsk asks = 1;
+}
+
+service API {
+    rpc Get(GetRequest) returns (GetReply) {}
+    rpc Query(QueryRequest) returns (QueryReply) {}
+}
```
Comment (asutula) [inline index/ask/pb/ask.proto]: Went ahead and made a service for the ask index. It's currently mixed in with the deals service. I plan on removing asks service code from the deals service in an upcoming PR.

Comment (jsign) [reply index/ask/pb/ask.proto]: :+1:

Comment (jsign) [reply index/ask/pb/ask.proto]: :+1:

**Pull request merged**